js/layout-javascript/agenda-code.js

```diff
@@ -1229,6 +1229,12 @@ DynamicList.prototype.renderLoopHTML = function () {
         _this.INCREMENTAL_RENDERING_BATCH_SIZE * (renderLoopIndex + 1)
       );
 
+      // Break render cycle after render to prevent rendering no results html
+      if (!nextBatch.length && renderLoopIndex > 0) {
+        resolve(_.flatten(_this.getAgendasByDay()));
+        return;
+      }
+
       var $renderBatch = $(template(nextBatch));
 
       $renderFull.add($renderBatch);
```

Comment:
A few lines beneath this line, there's a `renderLoopIndex === -1` case. I don't see why this would ever happen.

It was introduced in https://github.com/Fliplet/fliplet-widget-dynamic-lists/commit/4ebf34987632c1f5ce1a81bc815cf2abc04b602d#diff-0b2a0d477388682185df49e28c1ee71eR1012 nearly a year ago. I don't think it's necessary. Let's remove it for the purpose of cleaning this up for better readability -- and make sure QA tests this thoroughly.

Reply:
Fixed